src/main/java/seedu/address/model/person/Name.java

```diff
@@ -10,13 +10,14 @@
 public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Names should only contain alphanumeric characters and spaces up to 160 characters, "
+                    + "and it should not be blank";
 
     /*
      * The first character of the address must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]{1,159}";
```

Comment:
Realistic in setting the max numbet of characters in a name to 160 with consideration to outlier names which may be quite long.